packages/vscode-extension/src/plugins/network/network-plugin.ts

```diff
@@ -1,18 +1,54 @@
 import http, { Server } from "http";
+import { spawn } from "child_process";
+import path from "path";
 import { commands, Disposable, window } from "vscode";
 import { WebSocketServer, WebSocket } from "ws";
 import { Devtools } from "../../project/devtools";
 import { ToolKey, ToolPlugin } from "../../project/tools";
 import { extensionContext } from "../../utilities/extensionContext";
+
+import { Logger } from "../../Logger";
 import { NetworkDevtoolsWebviewProvider } from "./NetworkDevtoolsWebviewProvider";
 
 export const NETWORK_PLUGIN_ID = "network";
 
+function startViteServer(): Promise<void> {
```

Comment:
not sure why is this here. I don't think we should be using server unless in development mode. Similarly how we do this for the main panel: in dev mode we use vite such that we get hot reload but when bundling the extension we just load the packaged bundle.